README.md

```diff
@@ -1,4 +1,4 @@
-# binder
+# `mybinder.org` user documentation
```

Comment:
This is a big big big big big deal for me. I've always felt confused about this repo. I'd love to see it renamed to something like this as well.

Reply:
Opened https://github.com/jupyterhub/binder/issues/272